connect/runtime/src/main/java/org/apache/kafka/connect/storage/KafkaOffsetBackingStore.java

```diff
@@ -325,11 +325,12 @@ public Future<Void> set(final Map<ByteBuffer, ByteBuffer> values, final Callback
         return producerCallback;
     }
 
-    protected final Callback<ConsumerRecord<byte[], byte[]>> consumedCallback = new Callback<ConsumerRecord<byte[], byte[]>>() {
-        @Override
-        public void onCompletion(Throwable error, ConsumerRecord<byte[], byte[]> record) {
-            ByteBuffer key = record.key() != null ? ByteBuffer.wrap(record.key()) : null;
-            ByteBuffer value = record.value() != null ? ByteBuffer.wrap(record.value()) : null;
+    protected final Callback<ConsumerRecord<byte[], byte[]>> consumedCallback = (error, record) -> {
+        ByteBuffer key = record.key() != null ? ByteBuffer.wrap(record.key()) : null;
+        ByteBuffer value = record.value() != null ? ByteBuffer.wrap(record.value()) : null;
+        if (value == null) {
+            data.remove(key);
```

Comment:
It's a little strange to keep the `value` initializer the way it was now. IMO this would be more readable:
```java
        if (record.value() == null)
            data.remove(key);
        else
            data.put(key, ByteBuffer.wrap(record.value()));

```